From 00cb936b4d945161b3e259a825a39612794e0bcb Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Fri, 22 Mar 2019 08:35:46 +0800
Subject: [PATCH 705/706] i2c: imx: include of_gpio.h for the compiling error

According to the mainline, to use slot GPIO helpers will incur the drop
commits of this header file reference, so our SDK recovery fixes has a
slight behind from main line codes, meanwhile I don't see anything side
effects to include this header file to avoid the following compiling error:

drivers/i2c/busses/i2c-imx.c: In function 'i2c_imx_init_recovery_for_layerscape':
drivers/i2c/busses/i2c-imx.c:1196:18: error: implicit declaration of function 'of_get_named_gpio'; did you mean 'fwnode_get_named_gpiod'? [-Werror=implicit-function-declaration]
  i2c_imx->gpio = of_get_named_gpio(np, "fsl-scl-gpio", 0);
                  ^~~~~~~~~~~~~~~~~
                  fwnode_get_named_gpiod

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/i2c/busses/i2c-imx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i2c/busses/i2c-imx.c b/drivers/i2c/busses/i2c-imx.c
index 28bb058a0cb0..bc3d3d3e7edf 100644
--- a/drivers/i2c/busses/i2c-imx.c
+++ b/drivers/i2c/busses/i2c-imx.c
@@ -38,6 +38,7 @@
 #include <linux/of.h>
 #include <linux/of_device.h>
 #include <linux/of_dma.h>
+#include <linux/of_gpio.h>
 #include <linux/pinctrl/consumer.h>
 #include <linux/platform_data/i2c-imx.h>
 #include <linux/platform_device.h>
-- 
2.17.1

